matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/algorithms/megolm/MXMegolmDecryption.kt

```diff
@@ -341,6 +342,46 @@ internal class MXMegolmDecryption(private val userId: String,
         return olmDevice.hasInboundSessionKeys(roomId, senderKey, sessionId)
     }
 
+    override fun shareKeysWithDevice(exportedKeys: MegolmSessionData?, deviceId: String, userId: String) {
+        exportedKeys ?: return
+        cryptoCoroutineScope.launch(coroutineDispatchers.crypto) {
+            runCatching { deviceListManager.downloadKeys(listOf(userId), false) }
+                    .mapCatching {
+                        val deviceInfo = cryptoStore.getUserDevice(userId, deviceId)
+                        if (deviceInfo == null) {
+                            throw RuntimeException()
```

Comment:
is it worth a dedicated exception with more information or an error code?

Reply:
Again, I copied the existing code style. A dedicated exception should be much better I agree

Reply:
Oh we are doing that in existing code? feels bad

Reply:
Also there is no indication that this method throws, could end up crashing somewhere